### PR TITLE
[WIP]Demo

### PR DESCRIPTION
**Pull request by cipepser**

### CODE REVIEW

File: core/runtime/src/state_type.rs [hunk 1]
```diff
@@ -236,4 +237,35 @@ impl TryFrom<Vec<u8>> for Approved {
         let mut buf = s;
         Approved::from_bytes(&mut buf)
     }
+}
+
+#[derive(Encode, Decode, Clone, Debug, Default, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct Text(Vec<u8>);
```
Comment (osuketh) [inline core/runtime/src/state_type.rs:243]: why the name is `Text`, the type is `Vec<u8>`?

File: example/invoice-dx/app/src/lib.rs [hunk 2]
```diff
@@ -13,21 +15,28 @@ use anonify_runtime::{
...
-pub const MAX_MEM_SIZE: usize = 100;
-pub const CIPHERTEXT_SIZE: usize = MAX_MEM_SIZE + 30;
-
-
+// pub const MAX_MEM_SIZE: usize = 100;
+// pub const CIPHERTEXT_SIZE: usize = MAX_MEM_SIZE + 30;
 
 impl_memory! {
-
+    (0, "Body", Text)
```
Comment (osuketh) [inline example/invoice-dx/app/src/lib.rs:28]: prefer `"Invoice"` rather than `"Body"`

File: example/invoice-dx/app/src/lib.rs [hunk 3]
```diff
@@ -13,21 +15,28 @@ use anonify_runtime::{
...
 }
 
 impl_runtime! {
-
+    #[fn_id=0]
+    pub fn send_invoice(
+        self,
+        _sender: UserAddress,
+        recipient: UserAddress,
+        body: Text
```
Comment (osuketh) [inline example/invoice-dx/app/src/lib.rs:37]: update here as well.

File: example/invoice-dx/app/src/lib.rs (hunk 2 above)
Comment (cipepser) [reply example/invoice-dx/app/src/lib.rs]: fix it.
81a0f80f9d1f5faf0cdf2b474a385769310387af

File: core/runtime/src/state_type.rs (hunk 1 above)
Comment (cipepser) [reply core/runtime/src/state_type.rs]: `Bytes` is suitable for use `Vec<u8>`, so merge `Text` to it.
12019f4cb2e34fe3de4097f0826fd6a6c02f7fb0

File: example/invoice-dx/app/src/lib.rs (hunk 3 above)
Comment (cipepser) [reply example/invoice-dx/app/src/lib.rs]: fixed.
12019f4cb2e34fe3de4097f0826fd6a6c02f7fb0

**Pull request merged**